### DISCUSSION TITLE
Suffix with API call

### DISCUSSION THREAD

**Discussion by Tibo-Mouss** [moyix/fauxpilot#188] [Q&A]
Hello ! 
I'm working on an intellij plugin for fauxpilot, and it seems to me that the suffix part of the call isn't working.
The reason I say this is because in tests with java code, the suffix given is usually just the "}" that is present at the end of the class, and in every generation that is longer than a line, the completion adds an extra "}" as if it didn't know about the "}" already present after :

```
class Simple{  
    public static void main(String args[]){  
        System.out.println("Hello Java");  
    }

    /Function that generates a random integer
    ** public Int gen_int() {
        return int randomNum = ThreadLocalRandom.current().nextInt(0, 1000);
    }
}**   //<-- this is generated and shouldn't be

}  
```

I did check that "}" was being sent and I hope I'm understanding the suffix parameter right, so that's the only conclusion that I've stumbled upon

**Comment by thakkarparth007** [comment]
Hey @Tibo-Mouss , yes you're correct that the suffix part does not get used by the model. That is because the codegen models do not handle suffix. There are some models that do handle suffix (e.g., santacoder), but they need to be supported via the python backend. I will try to get that supported via a PR soon, but note that the python backend is going to be a bit slow. I'm not aware of Santacoder being supported by FasterTransformer yet.

**Reply by pai4451** [reply]
Hi @thakkarparth007, is the backend support suffix available now?

**Comment by spew** [comment]
I came here with the same question, although I would like to know more about the suffix parameter - does it actually get fed into the model and used in the inference when using OpenAI's API? I was imagining that it would enable a "Fill in the Middle" model to be used?

note: @Tibo-Mouss I was able to get a very crude suffix working by feeding the next line in as a 'stop' phrase through the stop parameter.
